zebra-chain/src/transaction/unmined.rs

```diff
@@ -174,55 +187,117 @@ pub struct UnminedTx {
 
     /// The size in bytes of the serialized transaction data
     pub size: usize,
+
+    /// The cost in bytes of the transaction, as defined in [ZIP-401].
+    ///
+    /// A reflection of the work done by the network in processing them (proof
+    /// and signature verification; networking overheads; size of in-memory data
+    /// structures).
+    ///
+    /// [ZIP-401]: https://zips.z.cash/zip-0401
+    pub cost: u32,
+}
+
+impl fmt::Display for UnminedTx {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        f.debug_struct("UnminedTx")
+            .field("transaction", &self.transaction)
+            .field("serialized_size", &self.size)
+            .finish()
+    }
 }
 
 // Each of these conversions is implemented slightly differently,
 // to avoid cloning the transaction where possible.
 
 impl From<Transaction> for UnminedTx {
     fn from(transaction: Transaction) -> Self {
+        let size = transaction
+            .zcash_serialized_size()
+            .expect("all transactions have a size");
+
         Self {
             id: (&transaction).into(),
-            size: transaction
-                .zcash_serialized_size()
-                .expect("all transactions have a size"),
+            size,
+            cost: std::cmp::max(size as u32, MEMPOOL_TRANSACTION_COST_THRESHOLD),
```

Comment:
Security: `as` silently truncates overflow. We should panic on overflow instead.

The transaction size check will be implemented in ticket #484 in sprint 20.

Can you also make this change for every other `cost` calculation?

```suggestion
            cost: std::cmp::max(size.try_into().expect("unexpectedly large size: should have already checked the transaction size limit"), MEMPOOL_TRANSACTION_COST_THRESHOLD),
```